### PR TITLE
feat(bin): enhance SHA384 extend utility with padding and tests

### PR DESCRIPTION
**Pull request by haraldh**
- Refactor `sha384-extend` to include digest padding and validation.
- Add `extend_sha384` function for hex-string-based digest extension.
- Introduce comprehensive test coverage for edge cases and errors.

### CODE REVIEW

File: bin/sha384-extend/src/main.rs [hunk 1]
```diff
@@ -1,39 +1,166 @@
...
+//! sha384-extend --base 01 26bb0c
+//! ```
 
 #![deny(missing_docs)]
 #![deny(clippy::all)]
 
 use anyhow::{Context, Result};
 use clap::Parser;
 use sha2::Digest;
+use teepot::pad;
```
[the remark this answers is not in the document]
Comment (haraldh) [reply bin/sha384-extend/src/main.rs:31]: yeah, was also my thought

File: crates/teepot/src/lib.rs [hunk 2]
```diff
@@ -17,14 +19,53 @@ pub mod server;
...
+/// // Error case: input too long
+/// let long_input = vec![1, 2, 3, 4, 5, 6];
+/// assert!(pad::<5>(long_input).is_err());
+/// # Ok(())
+/// # }
+/// ```
+///
+/// # Type Parameters
+/// * `T` - The fixed size of the output array in bytes
+pub fn pad<const T: usize>(input: Vec<u8>) -> anyhow::Result<[u8; T]> {
```
[the remark this answers is not in the document]
Comment (haraldh) [reply crates/teepot/src/lib.rs:52]: will do

File: bin/sha384-extend/src/main.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (haraldh) [reply bin/sha384-extend/src/main.rs]: copied

File: crates/teepot/src/lib.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (haraldh) [reply crates/teepot/src/lib.rs]: copied

File: bin/sha384-extend/src/main.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (haraldh) [reply bin/sha384-extend/src/main.rs]: hmm, but `teepot` is exactly that `pot` of utility functions used by TEE apps :)

[the remark this answers is not in the document]
Comment (haraldh) [reply bin/sha384-extend/src/main.rs]: moved to `teepot::util::pad`

**Review by LLBlumire — APPROVED** [review]
Lgtm
